forte/data/vocabulary.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=line-too-long
+from typing import List, Tuple, Dict, Union, Hashable, Iterable
+
+
+class Vocabulary:
+    """This class will store "Elements" that are added, assign "Ids" to them and
+    return "Representations" if queried. These three are the main concepts in this class.
+
+    1. Element: Any hashable instance that the user want to store.
+    2. Id: Each element will have an unique Id, which is an interger.
+    3. Representation: according to the configuration, the representation for an element
+        could be an interger (in this case, would be "Id"), or an one-hot vector (in this
+        case, would be a list of interger).
+
+    There are two special elements.
+
+    1. One is <PAD> element, which will be mapped into Id of 0 or -1 and have different
+        representation according to different setting.
+    2. The other one is <UNK> element, which, if added into the vocabulary, will be the
+        default element if the queried element is not found.
+
+    Here is a table on how our Vocabulary class behavior under different settings.
+
+    User_request = {
+        "vocab method":  "raw"(handle outside), "indexing",      "indexing",          "one-hot",                    "one-hot"
+        "need_pad":           assume False,        True,            False,              True,                         False
+    }
+
```

Comment:
https://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html

How about using real table syntax to document this.

Reply:
Got it.

Reply:
Is there a way to preview the `doc` so that I can see how it looks like?

Reply:
https://github.com/asyml/forte/blob/master/.travis.yml#L64 follow the scripts starting from here.

Reply:
Thank you very much.